### DISCUSSION TITLE
Setting material stiffness

### DISCUSSION THREAD

**Discussion by felipebandeiramm** [Dlubal-Software/RFEM_Python_Client#211] [Q&A] [answered]
Hello, I wonder how can I specify the modulus of elasticity to a user-defined material created in Python. I see the material.BasicProperties but only in JavaScript.

**Comment by dogukankaratas** [comment, marked as answer]
Hi,

Thanks for reaching out.

User defined material is still under work and we're hoping to add it to the library soon. 

Maybe I can offer an alternative solution for that. As you know importing and exporting SAF files is possible with the WebServices. So, if you define a material with values in a SAF file and import it via WS, it enables you to create user defined material.

![image](https://user-images.githubusercontent.com/61163577/209328537-7317c8b1-e704-449d-b26b-48cc187b2578.png)

It's a bit hard way to do it, but until the new release, this is the way to do it.

Feel free to reach out for other questions.

**Comment by scja1991** [comment]
Could you provide a minimal example? When I try to import a SAF-file RSTAB always generates a new model. But my goal is to change the matterialsettings of an existing model.

**Reply by OndraMichal** [reply]
Hi,
first open model/file with defined materials and then run your code on it. That is how we test properties which are needed for testing but are not yet implemented in WS as object or function.

**Comment by JesperWB** [comment]
Are there any updates on this topic? Is the user defined material in RFEM6 currently possible using python?

**Reply by OndraMichal** [reply]
Hi,
I added UserDefinedMaterial() function to Material class (PR#331, soon should be in main branch) so every parameter can be asigned. Please check it there.
